[K7] Side nav interaction adjustments

ryankeairns opened the pull request:
📓 This PR previously contained some initial work on adding a 'pin collapsed' button, but due to time and underlying complexity (mostly on the design/interaction side), we need to push that off. That said, the seemingly minor adjustments made by this PR provide some immediate relief for users who have reported inadvertent hovers coupled with a slow-to-close nav.

## Summary

This PR adjust the K7 side nav interaction in the following ways based upon internal feedback and personal observations:

- increase delay before expanding (prevent nav expanding if I accidentally/momentarily move over it)
- remove delay before collapsing (when I mouse out, collapse the nav more immediately)
- increase speed of the collapse animation (...and do it quickly)

~This PR also adds a new footer link to the bottom of the side nav. This will provide users the to keep the side nav in a collapsed state.~ This has been deferred, see note above.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
Pinging @elastic/kibana-design

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10771/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10779/)

ryankeairns commented:
@legrego Larry, here is the K7 side nav PR where we're making some tweaks. There is no EUI build yet, so to run this you have to link EUI. Also, the 'pin closed' feature has not yet been hooked up... but you can follow along here :)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10864/)

cchaos commented on src/ui/public/chrome/directives/header_global_nav/components/header.tsx:
```diff
@@ -254,6 +256,7 @@ class HeaderUI extends Component<Props, State> {
               'navDrawer',
               this.state.isCollapsed ? 'collapsed' : 'expanded'
             )}
+            style={{ transitionDuration: '90ms', transitionDelay: '150ms' }} // TODO Change defaults in EUI
```
I think we've been told we can no longer use `style` tags directly on components for security reasons. Am I correct @chandlerprall ?

ryankeairns replied:
I'll remove this if we squeeze in one more EUI build and Kibana update before FF.

ryankeairns replied:
I'm curious to hear that answer, regardless, but I'll be removing this from the PR as a change is coming on the EUI side.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11012/)

ryankeairns commented:
The inline styles have been removed. If you pull this now then you won't see the quick collapse animation. I'll rebase and correct that once https://github.com/elastic/kibana/pull/30116 is merged.

cjcenizal commented:
It sounds like we're all mostly aligned, I just wanted to echo @alexfrancoeur's impressions and add that we should probably be concerned about users possibly perceiving the removal of pinning as a regression, since the UI has offered this function since 5.0. Something to keep an eye on as we gather feedback during the beta period.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11070/)

ryankeairns commented:
I've opened a new issue to track ongoing work that addresses the feedback above. More improvements to come for the new side nav - https://github.com/elastic/kibana/issues/30121
